src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elementsData;
+    private int size;
+
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
```

Comment:
redundant `this` bacause there are no other variables with such a name in scope and `size` has a default value as it is a primitive field
```suggestion
    public ArrayList() {
        elementsData = (T[]) new Object[DEFAULT_CAPACITY];
    }
```